packages/khalti/test/src/khalti_http_client_test.dart

```diff
@@ -0,0 +1,259 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:khalti/khalti.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+
+typedef Caller = Future<HttpResponse> Function(String, Map<String, Object>);
+
+class _MockHttp extends Mock implements http.Client {}
+
+class _MockResponse extends Mock implements http.Response {}
+
+class _FakeUri extends Fake implements Uri {}
+
+void main() {
+  late _MockHttp _mockHttp;
+  late KhaltiHttpClient _khaltiHttpClient;
+
+  setUp(
+    () {
+      _mockHttp = _MockHttp();
+      _khaltiHttpClient = KhaltiHttpClient(httpClient: _mockHttp);
+    },
+  );
```

Comment:
Since the client doesn't have its own state. So, you should be able to use same instance for all the tests, without having to use `setUp`